### PR TITLE
refactor: clean and improve test/framework

### PR DESCRIPTION
**Pull request by bartsmykla**
- Moved logic of starting tcp and udp servers from the netns
  to appropriate {tcp,udp}/server.go files
- Moved out logic to close tcp connections from the current,
  predefined callbacks (it can be necessary in the future, when
  we will want to test invalid tcp packets dropping)
- Fixed/imroved test entries generation as there was potential
  bug (edge case) when there was clash in generated random, free
  ports
- Refactored tcp server as there was a bug where we were spawning
  separate goroutines inside locked os, which changed the server
  logic and it's currently prepared to handle only one connection
  (which is fine for our current, test purposes)

### CODE REVIEW

File: test/blackbox_tests/dns_test.go [hunk 1]
```diff
@@ -20,27 +20,29 @@ var _ = Describe("Outbound DNS/UDP traffic to port 53", func() {
 	var ns *netns.NetNS
 
 	BeforeEach(func() {
-		DeferCleanup(ns.Cleanup)
-
 		ns, err = netns.NewNetNSBuilder().Build()
 		Expect(err).To(BeNil())
 	})
 
+	AfterEach(func() {
```
Comment (lobkovilya) [inline test/blackbox_tests/dns_test.go:27]: I was thinking we decided to go with Defer? :)

Comment (bartsmykla) [reply test/blackbox_tests/dns_test.go:27]: We did, but it somehow was not reliable in these tests, and I didn't had enough time to debug why, as moving cleaning to AfterEach fixed all issues.

**Review by slonka — APPROVED** [review]
Overall looks good, one nitpick around the random generation

File: test/blackbox_tests/dns_test.go [hunk 2]
```diff
@@ -59,11 +61,17 @@ var _ = Describe("Outbound DNS/UDP traffic to port 53", func() {
...
-			ports := socket.GenerateRandomPorts(50, consts.DNSPort)
-
 			var entries []TableEntry
-			for port := range ports {
-				entries = append(entries, Entry(EntryDescription("to port %d, from port 53"), port))
+			lockedPorts := []uint16{consts.DNSPort}
+
+			for i := 0; i < 50; i++ {
+				randomPorts := socket.GenerateRandomPortsSlice(1, lockedPorts...)
+				// This gives us more entropy as all ports generated will be
```
[the remark this answers is not in the document]
Comment (bartsmykla) [reply test/blackbox_tests/dns_test.go]: 1. Generate 50 random ports during entries generation: `[1, 51, 952, 2343, ..., 81]`
2. BeforeEach: generate random TCP port: `2343`
3. Run test and `GOTO #2` 50 times

BeforeEach doesn't know about the 50 ports from #1 and as they are not locked in any way, there is a possibility that tcp server port will be the same as the randomly generated port for test.
In this case it's painless, but in the tests with port exclusion there was a possibility that tcp server port and exclude ports were the same, which would make these tests flaky (in very rare edge case)

File: test/blackbox_tests/dns_test.go (hunk 1 above)
Comment (bartsmykla) [reply test/blackbox_tests/dns_test.go:27]: ok, I know why - `DeferCleanup(ns.Cleanup)` is always being called on `nil` as it's "locking" the `ns` here
